### PR TITLE
Feature/windows installer

### PR DESCRIPTION
**Pull request by julianbermudez**
Enable Windows installer build using NSIS as install builder.
It is compliant with the Fast RTPS installer architecture.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -52,10 +52,15 @@ option(EPROSIMA_BUILD "Activate internal building" OFF)
 option(EPROSIMA_BUILD_TESTS "Activate the building tests" OFF)
 option(THIRDPARTY "Activate the build of thirdparties" OFF)
 option(VERBOSE "Use verbose output" OFF)
+option(BUILD_INSTALLER "Activate the creation of a build to create windows installer" OFF)
```
Comment (richiware) [inline CMakeLists.txt]: In other project we use EPROSIMA_INSTALLER. Can you rename this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -52,10 +52,15 @@ option(EPROSIMA_BUILD "Activate internal building" OFF)
...
+
+if(BUILD_INSTALLER)
+    set(EPROSIMA_BUILD ON)
+endif()
 
 if(EPROSIMA_BUILD)
     set(THIRDPARTY ON)
-    set(EPROSIMA_BUILD_TESTS ON)
+    #    set(EPROSIMA_BUILD_TESTS ON)
```
Comment (richiware) [inline CMakeLists.txt]: why do you remove tests?

File: cmake/packaging/eProsimaPackaging.cmake [hunk 3]
```diff
@@ -35,7 +35,7 @@ set(CPACK_RESOURCE_FILE_LICENSE "${PROJECT_SOURCE_DIR}/LICENSE")
 ###############################################################################
 # Create CMake package config file
 ###############################################################################
-if(NOT((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
+if(NOT((MSVC OR MSVC_IDE) AND BUILD_INSTALLER))
     set(DIR_EXTENSION "")
     if(EPROSIMA_INSTALLER_MINION)
```
Comment (richiware) [inline cmake/packaging/eProsimaPackaging.cmake]: EPROSIMA_INSTALLER_MINION not necessary here. It is used when the installer create on Windows multiple platform libraries.

File: cmake/packaging/windows/InstallRedistributables.nsh [hunk 4]
```diff
@@ -14,114 +14,43 @@
...
-
 Function InstallRedistributables
 
-    StrCpy $X64VS2013Needed "1"
-    StrCpy $I86VS2013Needed "1"
-    StrCpy $X64VS2015Needed "1"
-    StrCpy $I86VS2015Needed "1"
-
-    # Check if it is necessary to install to x64VS2013
+    # Check if it is necessary to install to redistributables.
```
Comment (richiware) [inline cmake/packaging/windows/InstallRedistributables.nsh]: This file is repited. Remove it.

File: CMakeLists.txt (hunk 1 above)
Comment (julianbermudez) [reply CMakeLists.txt]: Done, EPROSIMA_INSTALLER is not enable in Windows.

File: CMakeLists.txt (hunk 2 above)
Comment (julianbermudez) [reply CMakeLists.txt]: It was a mistake.

File: cmake/packaging/eProsimaPackaging.cmake (hunk 3 above)
Comment (julianbermudez) [reply cmake/packaging/eProsimaPackaging.cmake]: It was removed.

File: cmake/packaging/windows/InstallRedistributables.nsh (hunk 4 above)
Comment (julianbermudez) [reply cmake/packaging/windows/InstallRedistributables.nsh]: Remove this file and other unnecessary files.

File: CMakeLists.txt [hunk 5]
```diff
@@ -66,6 +71,37 @@ if(MSVC OR MSVC_IDE)
...
+###############################################################################
+
+# Install path
+set(BIN_INSTALL_DIR bin/ CACHE PATH "Installation directory for binaries")
+set(INCLUDE_INSTALL_DIR include/ CACHE PATH "Installation directory for C++ headers")
+set(LIB_INSTALL_DIR lib/ CACHE PATH "Installation directory for libraries")
+set(DATA_INSTALL_DIR share/ CACHE PATH "Installation directory for data")
+set(DOC_INSTALL_DIR ${DOC_DIR} CACHE PATH "Installation directory for documentation")
+if(MSVC OR MSVC_IDE)
+    set(LICENSE_INSTALL_DIR share/${PROJECT_NAME} CACHE PATH "Installation directory for licenses")
```
Comment (richiware) [inline CMakeLists.txt]: This can be joined. Both produce the same directory.

Comment (richiware) [reply CMakeLists.txt]: Maybe check where are the directories where Fast RTPS installs the License.

File: CMakeLists.txt [hunk 6]
```diff
@@ -240,7 +256,7 @@ install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/micrortps/agent/
 
 # Export library
 install(EXPORT ${PROJECT_NAME}Targets
-    DESTINATION ${LIB_INSTALL_DIR}/${PROJECT_NAME}/cmake
+    DESTINATION share/${PROJECT_NAME}/cmake
```
Comment (richiware) [inline CMakeLists.txt]: Use DATA_INSTALL_DIR.

File: CMakeLists.txt [hunk 7]
```diff
@@ -270,5 +292,79 @@ configure_package_config_file(${PROJECT_SOURCE_DIR}/cmake/packaging/Config.cmake
     PATH_VARS BIN_INSTALL_DIR INCLUDE_INSTALL_DIR LIB_INSTALL_DIR
     )
 install(FILES ${PROJECT_BINARY_DIR}/cmake/config/${PROJECT_NAME}Config.cmake
-    DESTINATION ${LIB_INSTALL_DIR}${DIR_EXTENSION}/${PROJECT_NAME}/cmake
+    DESTINATION share/${PROJECT_NAME}/cmake
```
Comment (richiware) [inline CMakeLists.txt]: Use DATA_INSTALL_DIR

File: cmake/common/eprosima_libraries.cmake [hunk 8]
```diff
@@ -55,7 +55,10 @@ macro(eprosima_find_package package)
                 foreach(opt_ ${FIND_OPTIONS})
                     set(${opt_} ON)
                 endforeach()
+                set(EPROSIMA_INSTALLER_TEMP ${EPROSIMA_INSTALLER})
```
Comment (richiware) [inline cmake/common/eprosima_libraries.cmake]: Please, add a comment to explain why we decided to do this. And update our subtree cmake-common

File: cmake/packaging/eProsimaPackaging.cmake [hunk 9]
```diff
@@ -35,11 +35,8 @@ set(CPACK_RESOURCE_FILE_LICENSE "${PROJECT_SOURCE_DIR}/LICENSE")
 ###############################################################################
 # Create CMake package config file
 ###############################################################################
-if(NOT((MSVC OR MSVC_IDE) AND EPROSIMA_INSTALLER))
+if(NOT((MSVC OR MSVC_IDE) AND BUILD_INSTALLER))
```
Comment (richiware) [inline cmake/packaging/eProsimaPackaging.cmake]: BUILD_INSTALLER?

File: CMakeLists.txt (hunk 5 above)
Comment (julianbermudez) [reply CMakeLists.txt]: Done!

File: CMakeLists.txt (hunk 6 above)
Comment (julianbermudez) [reply CMakeLists.txt]: Done!

File: cmake/common/eprosima_libraries.cmake (hunk 8 above)
Comment (julianbermudez) [reply cmake/common/eprosima_libraries.cmake]: Added

**Pull request merged**